test/test_cross_sections.py

```diff
@@ -174,3 +174,17 @@ def test_constant_lat_out_of_bounds(self):
         )
 
         assert len(candidate_faces) == 0
+
+
+class TestExtremes:
+    def test_extreme_latitude(self):
+        node_lon = np.array([-40, -40, 40, 40])
+        node_lat = np.array([-20, 20, 20, -20])
+        face_node_connectivity = np.array([[0, 1, 2, 3]], dtype=np.int64)
+
+        uxgrid = ux.Grid.from_topology(node_lon, node_lat, face_node_connectivity)
+
+        out1 = uxgrid.get_faces_at_constant_latitude(lat=20)
+        out2 = uxgrid.get_faces_at_constant_latitude(lat=20)
```

Comment:
How about ? 
```
-        out2 = uxgrid.get_faces_at_constant_latitude(lat=20)
+        out2 = uxgrid.get_faces_at_constant_latitude(lat=-20)
```

Also, what means extreme here? along the edges of the face? a small doc string would be good.

Reply:
Thanks for pointing this out, I made a typo in this tests.

Since we treat each edge as an arc, using the spherical bounds should yield the same result for 20 degrees and 21 degrees, which would otherwise fail if we didn't handle the spherical geometry.

Reply:
Doing some trial and error, I found that +/-25.41 still intersects but not 25.42

Reply:
These are the bounds for the single face, computed using Hongyu's implementation. The maximum along the arc will be greater than 20 degrees since we treat it as a great circle arc. 

![image](https://github.com/user-attachments/assets/61a4f7c1-06a8-4909-80eb-32e33f928c0c)